### DISCUSSION TITLE
Form component should implement Countable for full Twig loop support

### DISCUSSION THREAD

**Discussion by johndwells** [solspace/craft-freeform#209] [Feature Requests]
When rendering a form manually, a typical setup involves:

```
{% set form = craft.freeform.form('formHandle') %}
{% for row in form %}
    ...
{% endfor %}
```

Inside the `for` loop, the variable `loop` is missing many of its special variables. According to Twig's documentation, this is because `Solspace\Freeform\Library\Composer\Components\Form` does not implement `Countable`:

> The loop.length, loop.revindex, loop.revindex0, and loop.last variables are only available for PHP arrays, or objects that implement the Countable interface. They are also not available when looping with a condition.

- https://twig.symfony.com/doc/2.x/tags/for.html#the-loop-variable

Implementing `Countable` is straightforward, all that needs adding is a `count()` method, like so:

```
    public function count() {
    	return count($this->currentPageRows);
    }
```

:pray:

**Comment by johndwells** [comment]
FWIW, here is the workaround I'm using in the meantime:

```
{% set rows = form.getCurrentPage().getRows() %}
{% for row in rows %}
   ...
{% endfor %}
```

**Reply by kjmartens** [reply]
Thanks for this. We will check into it. :)
